mlflow/store/tracking/file_store.py

```diff
@@ -515,13 +515,7 @@ def _get_metric_history(self, run_info, metric_key):
     @staticmethod
     def _get_param_from_file(parent_path, param_name):
         _validate_param_name(param_name)
-        param_data = read_file_lines(parent_path, param_name)
-        if len(param_data) > 1:
-            raise Exception("Unexpected data for param '%s'. Param recorded more than once"
-                            % param_name)
-        # The only cause for param_data's length to be zero is the param's
-        # value is an empty string
-        value = '' if len(param_data) == 0 else str(param_data[0].strip())
+        value = read_file(parent_path, param_name)
```

Comment:
Quick sanity-check: What's the behavior when the FileStore writes the same parameter twice with different values? 

eg
```
with mlflow.start_run() as mlrun:
    mlflow.log_param("foo", "bar")
    mlflow.log_param("foo", "baz")
```

will the resulting file for "foo" be appended to or overwritten? The former would be problematic since read_file will return both values.

It might be good to add a unit test for this scenario since we no longer raise an obvious error ("Param recorded more than once") in that case.

Reply:
I believe it should be the latter case since the function used (correct me if I'm wrong) to write the param truncates the param value when writing:
```
def write_to(filename, data):
    with codecs.open(filename, mode="w", encoding=ENCODING) as handle:
        handle.write(data)
```

I think writing a test for this case would be great though!

Reply:
@AveshCSingh @juntai-zheng Great catches! Parameter values are meant to be immutable. Accordingly, I've introduced validation logic to ensure that multiple writes for the same parameter key with different values are rejected.